### PR TITLE
Feature/ec-cc-pages

### PR DESCRIPTION
**Pull request by noahmanger**
Adds basic itemized electioneering and communication costs pages.

The CC endpoint is currently throwing 500s, so I'm not able to see if it works, but this gets the basic structure in place.

To do / dependencies:
- [x] Finalize columns, filters and detail panel rows in https://github.com/18F/openFEC-web-app/issues/1046
- [x] Update endpoints based on these needs (specific issues forthcoming)
- [x] Revise filters, columns and detail panel based with the new data
- [x] Pull in filter-tags markup changes after #1015 is merged in

### CODE REVIEW

File: templates/macros/filters/years.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{% macro years(name, label, maxYear=2016, minYear=1978) %}
```
Comment (jmcarp) [inline templates/macros/filters/years.html]: Use camel_case for python variable names (jinja macros count as python).

File: static/templates/electioneering-communications.hbs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <td class="panel__term">Office</td>
+      <td class="panel__data">{{ candidate_office }}</td>
+    </tr>
+    {{#if candidate_office_state }}
+    <tr>
+      <td class="panel__term">State</td>
+      <td class="panel__data">{{ candidate_office_state }}</td>
+    </tr>
+    {{/if}}
+    {{#if candidate_office_district }}
```
Comment (jmcarp) [inline static/templates/electioneering-communications.hbs]: Null districts can be "00" and may still show up.

File: static/templates/electioneering-communications.hbs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    </tr>
+    <tr>
+      <td class="panel__term">Party</td>
+      <td class="panel__data">{{ candidate_party_affiliation }}</td>
+    </tr>
+    <tr>
+      <td class="panel__term">Office</td>
+      <td class="panel__data">{{ candidate_office }}</td>
+    </tr>
+    {{#if candidate_office_state }}
```
Comment (jmcarp) [inline static/templates/electioneering-communications.hbs]: Null states can be "US" and still show up.

File: static/templates/communication-costs.hbs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    <tr>
+      <td class="panel__term">Report year</td>
+      <td class="panel__data">{{ report_year }}</td>
+    </tr>
+    <tr>
+      <td class="panel__term">Transaction date</td>
+      <td class="panel__data">{{ datetime transaction_date pretty=true }}</td>
+    </tr>
+    <tr>
+      <td class="panel__term">Communication class</td>
```
Comment (LindsayYoung) [inline static/templates/communication-costs.hbs]: I am not sure we need to display communication class. Right now, we are showing  codes, and the options are M, E or S which I think is membership, executive personnel or stockholders. If we want to display this information, we will want to decode it. Also, it is blank in a third of the data.

File: static/templates/communication-costs.hbs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      <td class="panel__data">{{ communication_class }}</td>
+    </tr>
+    <tr>
+      <td class="panel__term">Communication type</td>
```
Comment (LindsayYoung) [inline static/templates/communication-costs.hbs]: If we display this we would want to decode it as well:
 DM - direct mail
 O - other
 TM - telegram!
 TP - telephone

**Comment by noahmanger** [comment]
Ok, so I'm sure the columns and such will continue to change, but this is now ready for review.

There's a couple dependencies on the API side before we merge:
- https://github.com/18F/openFEC/issues/1486
- https://github.com/18F/openFEC/issues/1485

But other than those, the table and panels are all working and I set up the links from the aggregate tables on election pages.

**Comment by noahmanger** [comment]
Would be nice to deploy this to dev sometime soon so that FEC can take a look at the data.

**Pull request merged**